src/amo/sagas/collections.js

```diff
@@ -338,6 +339,10 @@ export function* deleteCollection({
       errorHandlerId: errorHandler.id,
       username,
     }));
```

Comment:
So it seems the `fetchUserCollectionsAction` does not remove the old/existing collections for the current logged-in user. It means deleting a collection and clicking the "back" button will load the previously deleted collection. I am not sure that's what we want.

Reply:
I think the issue is actually with `loadUserCollections`, not `fetchUserCollections` in that it loads all the newly found collections, but adds them to any existing collections. I believe that is because of this line: https://github.com/mozilla/addons-frontend/blob/master/src/amo/reducers/collections.js#L968. If we were to remove that, then `loadUserCollections` would always load a fresh set of collections, which seems like what we want to do. Do you have any idea why the code might have been written that way to begin with? When might we want to just add to an existing set of user collections?

Reply:
The line 968 of the `collections` reducer modifies a `userCollections` attribute in the state, but when I load a collection page directly, this attribute is an empty object:

![screen shot 2018-06-05 at 19 17 09](https://user-images.githubusercontent.com/217628/40991764-16fc9532-68f5-11e8-97db-c322517ffe64.png)

It is only after I delete the collection that this `userCollections` attribute gets populated.

Reply:
Isn't it `loadCollectionIntoState()` that always keep adding data to the state?

Reply:
Sorry, @willdurand, I am getting confused by this discussion. I'm now unclear about what you think the issue actually is. Is it as simple as making sure that a collection gets unloaded from state when we delete it, or is there more to it than that?

In either case, do you think we can make this a follow-up issue because it doesn't really interfere with the delete feature and only represents an annoyance for an edge-case?

Reply:
The issue is that we don't unload a deleted collection. I think it could be done in this patch (because the diff is rather small and also because I noticed this bug while reviewing this patch). Let me know if you prefer a follow-up issue though.

Reply:
Ok, I've pushed a commit which unloads the collection after deleting it. Could you please take another look?